controllers/operatorconfigmap_controller.go

```diff
@@ -591,3 +598,62 @@ func labelClientOperatorSubscription(c *OperatorConfigMapReconciler) error {
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		// empty if annotation doesn't exist or else gets desired channel
+		channel := storageClients.Items[idx].GetAnnotations()[utils.DesiredSubscriptionChannelAnnotationKey]
+		// skip clients with no/empty desired channel annotation
+		if channel != "" {
+			// check if we already established a desired channel
+			if desiredChannel == "" {
+				desiredChannel = channel
+			}
+			// check for agreement between clients
+			if channel != desiredChannel {
+				desiredChannel = ""
+				// two clients didn't agree for a same channel and no need to continue further
+				break
+			}
+		}
+	}
+	if desiredChannel == "" {
+		// no desired channel specified or no agreement between clients
+		return nil
+	}
+
+	subscriptions := &opv1a1.SubscriptionList{}
+	err := c.list(subscriptions, client.InNamespace(c.OperatorNamespace), client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue}, client.Limit(1))
```

Comment:
Long line please split
```suggestion
	err := c.list(
		subscriptions, 
		client.InNamespace(c.OperatorNamespace), 
		client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue}, 
		client.Limit(1),
	)
```

Reply:
done.